### PR TITLE
Payloads

### PR DESCRIPTION
**Pull request by jdonszelmann**
Added 3 original py-ipv8 payloads to test against. (headers and hash removed as that's not supported yet).

Optimized some code and made a benchmark for the decoding process.

### CODE REVIEW

File: src/networking/payloads/binmemberauthenticationpayload.rs [hunk 1]
```diff
@@ -0,0 +1,41 @@
+use serde::{Serialize,Deserialize};
+use crate::networking::payloads::Ipv8Payload;
+use crate::networking::serialization::varlen::VarLen16;
+
+#[derive(Debug, PartialEq, Serialize, Deserialize)]
+pub struct BinMemberAuthenticationPayload {
+  /// The global time in the system. Uses a lamport clock system.
```
Comment (NULLx76) [inline src/networking/payloads/binmemberauthenticationpayload.rs]: This comment is out of place

File: src/networking/serialization/mod.rs [hunk 2]
```diff
@@ -97,6 +111,28 @@ mod tests {
     // doesnt have anything but needed for the default implementation (as of right now)
   }
 
+// only works with feature(test) and with `extern crate test; use test::Bencher`
+//  #[bench]
```
Comment (NULLx76) [inline src/networking/serialization/mod.rs:127]: We probably need to decide which testing framework to use libtest or something like Criterion.

File: tests/testpyipv8packets.rs [hunk 3]
```diff
@@ -0,0 +1,151 @@
...
+      },
+      source_lan_address: Address {
+        address: Ipv4Addr::new(192,168,1,75),
+        port: 8090
+      },
+      source_wan_address: Address {
+        address: Ipv4Addr::new(192,168,1,75),
+        port: 8090
+      },
+      // TODO i am not entirely sure of the extraction of bits is 100% right. though public does seem plausible.
```
Comment (NULLx76) [inline tests/testpyipv8packets.rs:140]: Important to figure this out

**Review by NULLx76 — CHANGES_REQUESTED** [review]

File: src/networking/serialization/mod.rs (hunk 2 above)
Comment (NULLx76) [reply src/networking/serialization/mod.rs:127]: The bench is only available in `nightly` therefore classified as "unstable" and after quickly looking into it stale as well.

**Review by dsluijk — APPROVED** [review]

File: src/networking/serialization/mod.rs (hunk 2 above)
Comment (NULLx76) [reply src/networking/serialization/mod.rs:127]: We decided to use the extenal benchmarker Criterion.rs

File: tests/testpyipv8packets.rs (hunk 3 above)
Comment (jdonszelmann) [reply tests/testpyipv8packets.rs:140]: something to keep in mind

**Review by NULLx76 — APPROVED** [review]

**Pull request merged**